src/CurveCallee.sol

```diff
@@ -0,0 +1,182 @@
+}
+
+interface CharterManagerLike {
+    function exit(address crop, address usr, uint256 val) external;
+}
+
+interface WstEthLike is TokenLike {
+    function unwrap(uint256 _wstEthAmount) external returns (uint256);
+    function stETH() external view returns (address);
+}
+
+interface CurveLike {
+    function exchange(int128 i, int128 j, uint256 dx, uint256 min_dy)
+        external returns (uint256 dy);
+}
+
+interface WethLike is TokenLike {
+    function deposit() external payable;
+}
+
+interface UniV3Like {
+    
+    struct Params {
+        bytes   path;
+        address recipient;
+        uint256 deadline;
+        uint256 amountIn;
+        uint256 amountOutMinimum;
+    }
+
+    function WETH9() external view returns (address);
+
+    function exactInput(UniV3Like.Params calldata params)
+        external payable returns (uint256 amountOut);
+}
+
+contract CurveCallee {
+    CurveLike               public curve;
+    UniV3Like               public uniV3;
+    DaiJoinLike             public daiJoin;
+    TokenLike               public dai;
+
+    uint256                 public constant RAY = 10 ** 27;
+
+    function add(uint x, uint y) internal pure returns (uint z) {
+        require((z = x + y) >= x, "ds-math-add-overflow");
+    }
+    function sub(uint x, uint y) internal pure returns (uint z) {
+        require((z = x - y) <= x, "ds-math-sub-underflow");
+    }
+    function divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = add(x, sub(y, 1)) / y;
+    }
+
+    constructor(
+        address curveAddr_,
+        address uniV3Addr_,
+        address daiJoinAddr_
+    ) public {
+        curve = CurveLike(curveAddr_);
+        uniV3 = UniV3Like(uniV3Addr_);
+        daiJoin = DaiJoinLike(daiJoinAddr_);
+        dai = daiJoin.dai();
+
+        dai.approve(daiJoinAddr_, uint256(-1));
+    }
+
+    receive() external payable {}
+
+    function _fromWad(address gemJoin, uint256 wad) internal view returns (uint256 amt) {
+        amt = wad / 10 ** (sub(18, GemJoinLike(gemJoin).dec()));
+    }
+
+    function clipperCall(
+        address sender,            // Clipper caller, pays back the loan
+        uint256 owe,               // Dai amount to pay back        [rad]
+        uint256 slice,             // Gem amount received           [wad]
+        bytes calldata data        // Extra data, see below
+    ) external {
+        (
+            address to,            // address to send remaining DAI to
+            address gemJoin,       // gemJoin adapter address
+            uint256 minProfit,     // minimum profit in DAI to make [wad]
+             uint24 poolFee,       // uniswap V3 WETH-DAI pool fee
+            address charterManager // pass address(0) if no manager
+        ) = abi.decode(data, (address, address, uint256, uint24, address));
+
+        address gem = GemJoinLike(gemJoin).gem();
+        require(keccak256(bytes(TokenLike(gem).symbol())) == keccak256("wstETH"), "CurveCallee: only-wsteth");
```

Comment:
Cheaper to check vs the gemJoin's ilk, i.e `GemJoinLike(gemJoin).ilk() == "XXX"`, but tbh I don't think we need this as we don't have this kind of validation for other callees (which can potentially also get wrong gemJoins from the user).
We assume the callee user is trusted and knows what he is doing.

Reply:
removed check.